src/main/java/at/porscheinformatik/sonarqube/licensecheck/gradle/GradleInvoker.java

```diff
@@ -0,0 +1,147 @@
+
+    private final static String BUILD_GRADLE = "build.gradle";
+    private static String GRADLE_EXEC;
+
+    private final File projectRoot;
+
+    private String userSettings;
+
+    // todo: use gradle tooling api if possible
+    GradleInvoker(String projectRoot) throws Exception {
+        this.projectRoot = new File(projectRoot);
+        File buildGradle = new File(projectRoot, BUILD_GRADLE);
+
+        if (!buildGradle.exists()) {
+            throw new Exception("no build.gradle found");
+        }
+
+        GRADLE_EXEC = resolveGradleExecutable(this.projectRoot);
+
+        userSettings = null;
+        CommandLine cmd = getCommandLineArgs();
+        if (cmd != null) {
+            if (cmd.hasOption("I")) {
+                userSettings = cmd.getOptionValue("I");
+            }
+        }
+    }
+
+    String invoke(String... gradleTasks) throws IOException, GradleInvokerException {
+
+        String[] command = resolveFullGradleCommand(gradleTasks);
+
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        Process process = processBuilder.command(command).directory(projectRoot).start();
+
+        InputStream errorStream = process.getErrorStream();
+        InputStream inputStream = process.getInputStream();
+        String stderr = getOutput(errorStream);
+        String stdout = getOutput(inputStream);
+
+        while (process.isAlive()) {
+        }
+        if (process.exitValue() != 0) {
+            LOGGER.error("Failed execution of gradle command {}", Arrays.toString(command));
+            LOGGER.error("Gradle stderr: {}", stderr);
+            throw new GradleInvokerException("Failed execution of gradle command ");
+        }
+
+        return stdout;
+    }
+
+    private String resolveGradleExecutable(File projectRoot) {
+        File gradlew = new File(projectRoot, "gradlew");
+
+        if (gradlew.exists()) {
+            LOGGER.info("Using {} wrapper with version {}",
+                gradlew.getAbsolutePath(),
+                resolveGradleVersion(gradlew.getAbsolutePath()));
+            return gradlew.getAbsolutePath();
+        } else {
+            String gradle = "gradle";
+            LOGGER.info("Using {} with version {}",
+                gradle,
+                resolveGradleVersion(gradle));
+            return gradle;
+        }
+    }
+
+    private String resolveGradleVersion(String exec) {
+        ProcessBuilder processBuilder = new ProcessBuilder(exec, "--version");
+        Process process = null;
+        try {
+            process = processBuilder.start();
+            return getOutput(process.getInputStream());
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private String[] resolveFullGradleCommand(String[] gradleCommands) {
+        String[] baseCommands;
+        if (userSettings == null) {
+            baseCommands = new String[]{GRADLE_EXEC, "-i"};
```

Comment:
You will want to add the `--no-daemon` switch to any invocations, otherwise the process will be in contention in case it is run with sonarqube gradle plugin on a daemonized gradle run.